OpenRA.Mods.Common/Traits/Buildings/Bridge.cs

```diff
@@ -323,95 +257,75 @@ void UpdateState()
-				var delay = initialDamage == DamageState.Undamaged || NeighbourIsDeadShore(neighbours[direction]) ?
-					0 : info.RepairPropagationDelay;
-
-				self.World.AddFrameEndTask(w => w.Add(new DelayedAction(delay, () =>
-					neighbours[direction].Repair(repairer, direction, onComplete))));
+				health.Resurrect(self, repairer);
+				killedUnits = false;
+				KillUnitsOnBridge();
 			}
+			else
+				health.InflictDamage(self, repairer, new Damage(-health.MaxHP), true);
 		}
 
 		void INotifyDamageStateChanged.DamageStateChanged(Actor self, AttackInfo e)
 		{
-			Do((b, d) => b.UpdateState());
-
-			// Need to update the neighbours neighbour to correctly
-			// display the broken shore hack
-			if (info.ShorePieces.Contains(type))
-				for (var d = 0; d <= 1; d++)
-					if (neighbours[d] != null && neighbours[d].neighbours[d] != null)
-						neighbours[d].neighbours[d].UpdateState();
-		}
-
-		void AggregateDamageState(Bridge b, int d, ref DamageState damage)
-		{
-			if (b.health.DamageState > damage)
-				damage = b.health.DamageState;
-			if (b.Hut == null && d >= 0 && b.neighbours[d] != null)
-				AggregateDamageState(b.neighbours[d], d, ref damage);
-		}
+			UpdateState();
 
-		// Find the worst span damage before other hut
-		public DamageState AggregateDamageState()
-		{
-			var damage = health.DamageState;
-			Do((b, d) => AggregateDamageState(b, d, ref damage));
-			return damage;
+			foreach (var neighbour in Neighbours)
+			{
+				neighbour.UpdateState();
+
+				// Need to update the neighbours neighbour to correctly
+				// display the broken shore hack.
+				if (neighbour.info.ShorePieces.Contains(type))
+					foreach (var n in neighbour.Neighbours)
+						if (n != this)
+							n.UpdateState();
+			}
 		}
 
-		public void Demolish(Actor saboteur, int direction, BitSet<DamageType> damageTypes)
+		void IBridgeSegment.Demolish(Actor saboteur, BitSet<DamageType> damageTypes)
 		{
-			var initialDamage = health.DamageState;
 			self.World.AddFrameEndTask(w =>
 			{
-				// Use .FromPos since this actor is killed. Cannot use Target.FromActor
+				if (self.IsDead)
+					return;
+
+				// Use .FromPos since this actor is dead. Cannot use Target.FromActor.
 				info.DemolishWeaponInfo.Impact(Target.FromPos(self.CenterPosition), saboteur);
 
 				self.Kill(saboteur, damageTypes);
 			});
-
-			// Destroy adjacent spans between (including) huts
-			if (direction >= 0 && Hut == null && neighbours[direction] != null)
-			{
-				var delay = initialDamage == DamageState.Dead || NeighbourIsDeadShore(neighbours[direction]) ?
-					0 : info.RepairPropagationDelay;
-
-				self.World.AddFrameEndTask(w => w.Add(new DelayedAction(delay, () =>
-					neighbours[direction].Demolish(saboteur, direction, damageTypes))));
-			}
 		}
 
 		void IRadarSignature.PopulateRadarSignatureCells(Actor self, List<(CPos Cell, Color Color)> destinationBuffer)
 		{
 			destinationBuffer.AddRange(radarSignature);
 		}
+
+		string IBridgeSegment.Type => "GroundLevelBridge";
```

Comment:
Consider extracting a constant for `"GroundLevelBridge"` to avoid string typing.